swarm/pss/snapshot_test.go

```diff
@@ -0,0 +1,433 @@
+package pss
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/simulations"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/network/simulation"
+	"github.com/ethereum/go-ethereum/swarm/pot"
+	"github.com/ethereum/go-ethereum/swarm/state"
+)
+
+// needed to make the enode id of the receiving node available to the handler for triggers
+type handlerContextFunc func(*adapters.NodeConfig) *handler
+
+// struct to notify reception of messages to simulation driver
+// TODO To make code cleaner:
+// - consider a separate pss unwrap to message event in sim framework (this will make eventual message propagation analysis with pss easier/possible in the future)
+// - consider also test api calls to inspect handling results of messages
+type handlerNotification struct {
+	id     enode.ID
+	serial uint64
+}
+
+var (
+	runNodes    = flag.Int("nodes", 0, "nodes to start in the network")
+	runMessages = flag.Int("messages", 0, "messages to send during test")
+
+	pof   = pot.DefaultPof(256) // generate messages and index them
+	topic = BytesToTopic([]byte{0x00, 0x00, 0x06, 0x82})
+	mu    sync.Mutex // keeps handlerDonc in sync
+	sim   *simulation.Simulation
+
+	handlerDone   bool // set to true on termination of the simulation run
```

Comment:
I see you still are using globals in the test. I've already given my opinion on this; I think it is risky design, and I have even gotten scorn from the golang irc channel for doing the same myself at some point (code smell, I believe they called it). At a minimum I would recommend creating a struct that can encapsulate the state and all methods accessing the state (`isDone()`, `setDone()`), which every relevant method gets passed. I will not recommend merging this PR without this amendment.

I would also recommend that we look at how to refactor this test to fit the action/trigger/expectation paradigm of `p2p/simulations.go:Simulation` but for my part that can be a separate PR after this.